src/components/router/routerHelpers.js

```diff
@@ -29,6 +51,105 @@ const baseName = (helpers.TEST_MODE && '/') || (helpers.DEV_MODE && '/') || dyna
 const basePath =
   (helpers.TEST_MODE && '/') || (helpers.DEV_MODE && '/') || window.location.pathname.split(helpers.UI_NAME)[0];
 
+/**
+ * ToDo: now that navigation is generated we can bypass and use route configuration directly
+ * Review removing "navigation" and using the route config directly.
+ */
+/**
+ * Generate navigation list, used to help in look ups for "missing products" and provide additional route details.
+ *
+ * @param {Array} config
+ * @returns {{path: string, default: boolean, component: Node, productParameter: string,
+ *     viewParameter: string, activateOnError: boolean, id: string, isSearchable: boolean, pathParameter: string}[]}
+ */
+const generateNavigation = (config = routesConfig) =>
```

Comment:
required:
lets go ahead and try to remove the necessity for "generateNavigation" entirely. as a subsequent commit restructure the helpers associated with pulling product and routing information